Sources/FluentKit/Properties/Parent+EagerLoad.swift

```diff
@@ -12,101 +12,95 @@ extension Parent: AnyEagerLoadable {
             return
         }
 
-        if let subquery = request as? ParentSubqueryEagerLoad<To> {
+        if let subquery = request as? SubqueryEagerLoad {
             self.eagerLoadedValue = try subquery.get(id: id)
         } else {
             fatalError("unsupported eagerload request: \(request)")
         }
     }
 }
 
+
+// MARK: - Eager Loadable
+
 extension Parent: EagerLoadable {
-    public func eagerLoad<Model>(to builder: QueryBuilder<Model>)
-        where Model: FluentKit.Model
-    {
-        builder.eagerLoads.requests[self.eagerLoadKey] = ParentSubqueryEagerLoad<To>(
-            key: self.$id.key
-        )
+    public func eagerLoad<Model>(to builder: QueryBuilder<Model>) where Model: FluentKit.Model {
+        builder.eagerLoads.requests[self.eagerLoadKey] = self.eagerLoadRequest
     }
 }
 
+// MARK: Private
 
-extension OptionalParent: AnyEagerLoadable {
-    var eagerLoadKey: String {
-        return "p:" + self.$id.key
-    }
+extension Parent {
+    internal final class SubqueryEagerLoad: EagerLoadRequest {
+        typealias Loader = (Database, Set<To.StoredIDValue>) -> EventLoopFuture<[To]>
+        typealias LoadedParent = (id: To.StoredIDValue, model: Parent<To>.EagerLoaded)
 
-    var eagerLoadValueDescription: CustomStringConvertible? {
-        return self.eagerLoadedValue
-    }
+        let key: String
+        let loader: Loader
 
-    func eagerLoad(from eagerLoads: EagerLoads) throws {
-        guard let request = eagerLoads.requests[self.eagerLoadKey] else {
-            return
+        var storage: [LoadedParent]
```

Comment:
Why is `storage` not a dictionary of `[To.StoredIdValue: LoadedParent]`? It's accessed consistently as a key-value store, but as an array it's using `O(n)` linear index searches instead of `O(1)` key lookups. `StoredIDValue` would have to conform to `Hashable`, but that's all.

Reply:
You can discuss this one with @Joannis, but according to him, using the array lookup is supposed to be more performant due to the way a dictionary is stored in memory.

Reply:
I find that difficult to believe - `firstIndex(where:)`, for example, is an `O(n)` linear search which includes the overhead of invoking the closure each time - an overhead which can be non-trivial in some cases.  Subscripting a dictionary is `O(1)`, though it certainly can come with its own overheads in some cases. @Joannis, can you elaborate?